src/debug_menu.cpp

```diff
@@ -121,99 +128,147 @@ class mission_debug
         static std::string describe( const mission &m );
 };
 
-static int player_uilist()
+typedef std::pair<uilist_entry, bool> uilist_entry_pair_t;
+
+/**
+ * Filter entries to be displayed in a menu.
+ * @param entries: A vector where each element is a pair consisting of (first) a menu entry and (second) boolean indicating whether to show the entry or not.
+ * @param display_all_entries: boolean used to override the entries in the @p entries vector. If `true` then all entries are shown (no matter what is their boolean state);
+ *  if `false` then the boolean of the menu entry is taken into account.
+ * @returns A filtered vector of menu entries, depending on the given parameters.
+ */
+static std::vector<uilist_entry> filter_uilist( const std::vector<uilist_entry_pair_t> &entries,
+        bool display_all_entries )
```

Comment:
These boolean flags everywhere are a bit cumbersome. Instead of filtering menus, you could've simply split them: a menu which items are always "legal" (essentially only two for now), and everything else for "cheating".